modules/programs/exa.nix

```
@@ -23,10 +11,51 @@ in {
 
     enableAliases = mkEnableOption "recommended exa aliases";
 
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "--group-directories-first" "--header" ];
+      description = ''
+        Extra command line options passed to exa.
+      '';
+    };
+
+    icons = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Display icons next to file names (<option>--icons</option> argument).
+      '';
+    };
+
+    git = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        List each file's Git status if tracked or ignored (<option>--git</option> argument).
+      '';
+    };
+
     package = mkPackageOption pkgs "exa" { };
   };
 
-  config = mkIf cfg.enable {
+  config = let
+    cfg = config.programs.exa;
+
+    cmd = escapeShellArgs ([ "exa" ] ++ optional cfg.icons "--icons"
+      ++ optional cfg.git "--git" ++ cfg.extraOptions);
+
+    aliases = {
+      # Use `command` instead of hardcoding the path to exa so that aliases don't
+      # go stale after a system update.
+      exa = "command ${cmd}";
+      ls = "exa";
+      ll = "exa -l";
+      la = "exa -a";
+      lt = "exa --tree";
+      lla = "exa -la";
+    };
```

```suggestion
    exaArgs = optional cfg.icons "--icons"
      ++ optional cfg.git "--git" ++ cfg.extraOptions);

    aliases = {
      # Use `command` instead of hardcoding the path to exa so that aliases don't
      # go stale after a system update.
      exa = "exa ${exaArgs}";
      ls = "exa";
      ll = "exa -l";
      la = "exa -a";
      lt = "exa --tree";
      lla = "exa -la";
    };
```
Try this. Not tested.